lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1091,10 +1090,30 @@ namespace Js
             }
             else
             {
-                return object && object->HasOwnProperty(propertyId);
+#ifdef DEBUG
+                // In debug builds, calculate expectedAnswer up-front so that
+                // we can Assert the fastpath gets the right result.
+                BOOL expectedAnswer = object && object->HasOwnProperty(propertyId);
+#endif
+
+                PropertyString *propString = requestContext->GetPropertyString(propertyId);
```

Comment:
>GetPropertyString [](start = 61, length = 17)

This goes and create a PropertyString if one is not present in that case type below won't match, should we have a different function which returns only if PropertyString is present.